blockchain/stake/staketx.go

```diff
@@ -1245,13 +1340,29 @@ func FindSpentTicketsInBlock(block *wire.MsgBlock) *SpentTicketsInBlock {
 // in which case an error is returned.
 func CreateRevocationFromTicket(ticketHash *chainhash.Hash,
 	ticketMinOuts []*MinimalOutput, revocationTxFee dcrutil.Amount,
-	revocationTxVersion uint16, params *chaincfg.Params) (*wire.MsgTx, error) {
+	revocationTxVersion uint16, params *chaincfg.Params,
+	prevHeaderBytes []byte, isAutoRevocationsEnabled bool) (*wire.MsgTx, error) {
 
 	// Validate that the provided ticket minimal outputs have a non-zero length.
 	if len(ticketMinOuts) == 0 {
 		return nil, stakeRuleError(ErrSStxNoOutputs, "no minimal outputs")
 	}
 
+	// If the automatic ticket revocations agenda is active, then the fee must be
+	// 0 and the version must be 2.
+	if isAutoRevocationsEnabled {
+		if revocationTxFee != 0 {
+			str := "fee must be zero when auto revocations is active"
+			return nil, stakeRuleError(ErrSSRtxInvalidFee, str)
+		}
+
+		if revocationTxVersion != TxVersionAutoRevocations {
```

Comment:
As mentioned in the other comment, I think its better to check for the specific version that is currently valid.  It may be likely that it still applies for future transaction versions but I think that should be accounted for when the associated consensus change comes along.

Reply:
I went back and forth on this but decided to update it to `>=`.  We should still be able to update this to not have the agenda check and only rely on the version check as long as no revocations with a version greater than 2 are accepted into a block before it activates (which should be highly likely).  And if that is the case, then it would be nice that using `>=` should be more future-proof and not require an update if the rule doesn't change in future revocation versions.

Reply:
I recall that there are similar inequalities for ssgen as well as coinbase for the treasury changes, and I raised the question at some point.  Seems like just consistency would be best, but maybe there just needs to be good notes about these somewhere as well.

Reply:
It is updated in other cases that are just _checking_ the version, but this is in `CreateRevocationFromTicket`, which creates a revocation, so the version must be `2`.  Although the majority of checks were updated to use `>=`, consensus is still enforcing that the version [must be exactly `2` with the updated rules](https://github.com/decred/dcrd/blob/1210cc2c4eb263409c3b8db8b913a18e1ccf3d33/blockchain/validate.go#L483-L491).  The other locations are just checking `>=` as a convenience to not need to update them in future versions.